osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -405,24 +406,45 @@ void endOperation()
             }
         }
 
-        private void onRoomUpdated()
+        private void onRoomUpdated() => Scheduler.Add(() =>
         {
-            // user mods may have changed.
-            Scheduler.AddOnce(UpdateMods);
-        }
+            if (client.Room == null)
+                return;
 
-        private void onLoadRequested()
-        {
-            Debug.Assert(client.Room != null);
+            Debug.Assert(client.LocalUser != null);
 
-            int[] userIds = client.CurrentMatchPlayingUserIds.ToArray();
+            UpdateMods();
+
+            if (client.LocalUser.State == MultiplayerUserState.Spectating
```

Comment:
Not a huge fan of this code path here. Feels like there already could be a race condition when changing states where `StartPlay` is called more than once.

Did you consider changing the `LoadRequested` event to also fire for spectators?

Reply:
Sure, I've applied the necessary changes to make this work, requires server-side change.